devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -1365,6 +1866,579 @@ def deleteCode(codeID) {
+		def endHour = endTime.hour
+		def endMin = endTime.minute
+		
+		// For weekDayBitMask, the order of days is Sat, Fri, Thu, Wed, Tue, Mon, Sun with Sat as the most significant bit
+		short weekDayBitMask = schSettings.weekDayBitMask
+		
+		state["setWeekDayStartTime$codeID"] = schSettings.startTime
+		state["setWeekDayEndTime$codeID"] = schSettings.endTime
+		state["setWeekDayBitMask$codeID"] = weekDayBitMask
+		
+		def setCmds = []
+		def getCmds = []
+		def daysOfWeek = []
+		// dayOfWeek - Sunday = 0, Monday = 1, Tuesday = 2, Wednesday = 3, Thursday = 4, Friday = 5, Saturday = 6
+		for (def i = 0; i <= 6; i++) {
+			if (isBitSet(weekDayBitMask, i)) {
+				daysOfWeek << i
+				setCmds << secure(zwave.scheduleEntryLockV3.scheduleEntryLockWeekDaySet(startHour: startHour, startMinute: startMin,
+					setAction: 1, scheduleSlotId: (i + 1), stopHour: endHour, stopMinute: endMin, userIdentifier: codeID, dayOfWeek: i))
+				getCmds << requestWeekDaySchedule((i + 1), codeID)
+			}
+		}
+		if (setCmds && getCmds) {
+			cmds = delayBetween(setCmds, 200) + ["delay 2000"] + delayBetween(getCmds, 200)
+			state["setWeekDayWeekDays$codeID"] = daysOfWeek
+		}
+		
+		log.debug "[DTH] 'getWeekDayScheduleSetCmd()' returning with commands := $cmds"
+		return cmds
+	}
+	return null
+}
+
+/**
+ * Returns the command(s) for setting year day schedule
+ *
+ * @return The command(s) for setting year day schedule
+ */
+private def getYearDaySetScheduleCmd(schSettings) {
+	if (device.currentValue("numberOfSlotsYearDay")) {
+		log.trace "[DTH] Executing 'getYearDaySetScheduleCmd()' by ${device.displayName} with schSettings := $schSettings"
+		def cmds = []
+		
+		def codeID = Integer.parseInt(schSettings.codeId, 10)
+		
+		def startDate = schSettings.startDate
+		def startTime = schSettings.startTime
+		def endDate = schSettings.endDate
+		def endTime = schSettings.endTime
+		
+		state["setYearDayStartDate$codeID"] = startDate
+		state["setYearDayStartTime$codeID"] = startTime
+		state["setYearDayStopDate$codeID"] = endDate
+		state["setYearDayStopTime$codeID"] = endTime
+		
+		def startDateTime = getYearDayTimeDetails(startDate, startTime)
+		def startHour = startDateTime.hour
+		def startMinute = startDateTime.minute
+		def startDay = startDateTime.day
+		def startMonth = startDateTime.month
+		def startYear = startDateTime.year.toString()
+		startYear = startYear.substring(2, startYear.length())
+		startYear = Integer.parseInt(startYear, 10)
+		
+		def endDateTime = getYearDayTimeDetails(endDate, endTime)
+		def stopHour = endDateTime.hour
+		def stopMinute = endDateTime.minute
+		def stopDay = endDateTime.day
+		def stopMonth = endDateTime.month
+		def stopYear = endDateTime.year.toString()
+		stopYear = stopYear.substring(2, stopYear.length())
+		stopYear = Integer.parseInt(stopYear, 10)
+		
+		cmds << secure(zwave.scheduleEntryLockV3.scheduleEntryLockYearDaySet(scheduleSlotId: 1, setAction: 1, startDay: startDay, startHour: startHour, startMinute: startMinute,
```

Comment:
As a general comment, you have a bunch of these functions where you break out all your arguments into separate assignments and then just use them once as arguments to a function call.

I'd recommend just referencing them directly instead of having the extra assignment.